### PR TITLE
Sva 293 add es lint and prettier config to api directory new

### PR DESCRIPTION
**Pull request by M4RC1N76**
- [Description](#description)
  
# Description

- Added a prettierrc.json to the /api/ directory
- Added a eslintrc.js config file to the /api/ directory
- I have created separet ticket SVA-393-switch-from-common-javascript-statements-to-ES6-import-statements
- Have verified the code re-formats upon save with Prettier formatting, and when running npx eslint . the linter runs correctly and provides errors. There will need to be a separate ticket to fix all issues in the linters, and some of the config rules might need updating depending on this.
- In the /api/ directory package.json, i've installed two additional dependencies, as I was having issues with eslint and prettier. Installing the plugin 'eslint-plugin-prettier' fixed these issues. (I suspect it is because i'm using Prettier v3.0, unlike the app directory which uses Prettier v2.8.8 - Kelsie)
- Fixes # (SVA-293 ES Lint and Prettier)

## Type of change

Please delete options that are not relevant

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [x] Breaking change(fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

# Testing locally

- Run npm install and ensure that Prettier works and run the command npx eslint . in the api directory to ensure that you can see the linting working. You should see 3608 problems in the results. This is expected and need to be addressed in a separate ticket.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [x] I have removed any unnecessary comments or console logging
- [] I have made corresponding changes to the documentation (if required)
- [] I have addressed accessibility, if needed
- [] I have followed best practices, e.g. NativeBase approaches and theming
- [] I have checked the app in dark mode, if making front-end design changes
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] I have updated the version numbers in `package.json` files in the [app](DEPLOYMENT.md#app-deployment) and/or [api](DEPLOYMENT.md#api-deployment-on-aws) directories as needed

### CODE REVIEW

**Comment by CatMac228** [comment]
Hey @M4RC1N76! I might be doing something wrong but when I try the npx eslint command I'm not seeing anything come back saying a number of errors like you mentioned in the testing locally part. High chance there might be an issue with my build though so will need someone else to double check theirs!

**Comment by M4RC1N76** [comment]
Hi Cat, are you using branch ending with ......new? There are two branches one ends with the word ....kelsie other one is .....new . The SVA-293.....-new is the correct one.
Hope this will help. I will also write a guide like David did with his ticket in Notion.
Kind Regards Marcin Borowski

Sent from Yahoo Mail on Android 
 
  On Wed, 11 Oct 2023 at 17:33, ***@***.***> wrote:   


Hey @M4RC1N76! I might be doing something wrong but when I try the npx eslint command I'm not seeing anything come back saying a number of errors like you mentioned in the testing locally part. High chance there might be an issue with my build though so will need someone else to double check theirs!

File: api/package.json [hunk 1]
```diff
@@ -14,16 +14,20 @@
   "dependencies": {
     "@bugsnag/js": "^7.20.2",
     "@bugsnag/plugin-express": "^7.19.0",
+    "@react-native-community/eslint-config": "^3.2.0",
```
Comment (roryf) [inline api/package.json:17]: Since these are all developer tools, they should be in `devDependencies`

**Review by roryf — CHANGES_REQUESTED** [review]
@M4RC1N76 could we add a `lint` script to `package.json` so we can do `npm run lint` rather than rely on `npx`?

I also couldn't get it to generate errors with just `npx eslint` but adding `"lint": "eslint .",` to package.json it worked (as in gave me a screen full of errors...)

I notice you added `lint-staged`, this is usually used in conjunction with [`husky`](https://typicode.github.io/husky/) to auto-lint on commits, was there an intent to add that too or did you see value in just `lint-staged`? I guess since it should only lint changed files, it could enforce style upgrades on touched files until we can fix all the errors, otherwise it's a bit pointless until that is done.

There is also an errant file `app/.watchman-cookie-Marcins-MacBook-Air.local-1111-177`

**Comment by roryf** [comment]
> I notice you added lint-staged, this is usually used in conjunction with [husky](https://typicode.github.io/husky/) to auto-lint on commits, was there an intent to add that too or did you see value in just lint-staged?

Not 100% sure how this will work with two sub-directories, I see `app` already installs husky hooks, never used it in this scenario before - maybe it is clever enough to perform two separate lint-staged commands!

File: api/package.json [hunk 2]
```diff
@@ -6,35 +6,49 @@
...
     "node-fetch": "^3.0.0",
     "nodemon": "^2.0.12",
+    "prettier": "^3.0.0",
     "vimeo": "^2.1.1"
   },
   "devDependencies": {
     "@faker-js/faker": "^8.0.2",
     "babel-cli": "^6.26.0",
     "babel-preset-env": "^1.7.0",
+    "@bugsnag/js": "^7.20.2",
```
Comment (roryf) [inline api/package.json:35]: Bugsnag is actually a runtime dependency, so does need to be in `dependencies` - `devDependencies` is essentially for tooling that is only used for local development or testing.

**Review by roryf — CHANGES_REQUESTED** [review]
Couple of packages moved from `dependencies` to `devDependencies` were incorrect, there is also still an errant file `app/.watchman-cookie-Marcins-MacBook-Air.local-1111-177` in the diff.

**Comment by roryf** [comment]
@M4RC1N76 I think this is superseded by Scottish-Tech-Army/Volunteer-api#4 and can be closed?

**Pull request closed without merging**